src/model.js

```diff
@@ -20,6 +23,29 @@ const userSchema = new mongoose.Schema({
   instructorData: { type: instructorSchema }
 });
 
+// Detects change to the password field, both password update and password creation, then hashes the password before persistance
+userSchema.pre("save", async function (next) {
+  if (this.isModified("password")) {
+    try {
+      const salt = await bcrypt.genSalt(SALT_ROUNDS);
+      this.password = await bcrypt.hash(this.password, salt);
+    } catch (err) {
+      next(err); // Pass any errors to the next middleware
+    }
+  } else {
+    next();
+  }
+});
+
+// Method to compare input password with hashed password in DB
```

Comment:
It occurs to me that this comment needs state what this method returns (e.g. if a boolean then `true` means that the passwords match and `false` means that they don't).